mongodbatlas/mongodbatlas.go

```diff
@@ -239,29 +256,66 @@ func SetUserAgent(ua string) ClientOpt {
 // BaseURL of the Client. Relative URLS should always be specified without a preceding slash. If specified, the
 // value pointed to by body is JSON encoded and included in as the request body.
 func (c *Client) NewRequest(ctx context.Context, method, urlStr string, body interface{}) (*http.Request, error) {
-	rel, err := url.Parse(urlStr)
+	u, err := c.BaseURL.Parse(urlStr)
 	if err != nil {
 		return nil, err
 	}
 
-	u := c.BaseURL.ResolveReference(rel)
+	buf, err := c.serializeRequestBody(body)
+	if err != nil {
+		return nil, err
+	}
+
+	req, err := http.NewRequest(method, u.String(), buf)
+	if err != nil {
+		return nil, err
+	}
 
-	buf := new(bytes.Buffer)
 	if body != nil {
-		err = json.NewEncoder(buf).Encode(body)
+		req.Header.Set("Content-Type", mediaType)
+	}
+	req.Header.Add("Accept", mediaType)
+	if c.UserAgent != "" {
+		req.Header.Set("User-Agent", c.UserAgent)
+	}
+	return req, nil
+}
+
+// serializeRequestBody returns an ReadWriter object containing the body of the http request
+func (c *Client) serializeRequestBody(body interface{}) (io.ReadWriter, error) {
+	var buf io.ReadWriter
+	if body != nil {
```

Comment:
this if should be done before calling this method and if you do that then it makes sense to call this method `newEncodedBody`